src/data-workspace/inputs/boolean-radios.js

```diff
@@ -1,103 +1,72 @@
 import i18n from '@dhis2/d2-i18n'
 import { Button, Radio } from '@dhis2/ui'
 import cx from 'classnames'
-import React from 'react'
-import { useField } from 'react-final-form'
+import React, { useEffect, useState } from 'react'
 import { useSetDataValueMutation } from '../../shared/index.js'
 import styles from './inputs.module.css'
-import { convertCallbackSignatures, InputPropTypes } from './utils.js'
+import { InputPropTypes, convertBooleanValue } from './utils.js'
 
-// ? Will this fail to reflect a value on the server if it's not exactly `true` or `false`?
-// todo: may need to handle that when mapping server values to form initial values, e.g.
-// Currently it's working okay
-// boolean: accepts 1, 0, 'true', 'false'
-// Ex: if (dv.valueType === boolean) { formValue = dv.value ... etc }
-// does `isEqual` prop help make 1/true and 0/false/'' equal?
 export const BooleanRadios = ({
-    fieldname,
-    form,
     deId,
     cocId,
     disabled,
     locked,
     onKeyDown,
     onFocus,
+    onBlur,
+    setValueSynced,
+    initialValue,
 }) => {
-    const useFieldWithPatchedOnFocus = (...args) => {
-        const { input, ...rest } = useField(...args)
-        return {
-            ...rest,
-            input: {
-                ...input,
-                onFocus: (...args) => {
-                    input.onFocus(...args)
-                    onFocus?.(...args)
-                },
-            },
-        }
-    }
-    const yesField = useFieldWithPatchedOnFocus(fieldname, {
-        type: 'radio',
-        value: 'true',
-        subscription: { value: true },
-    })
-
-    const noField = useFieldWithPatchedOnFocus(fieldname, {
-        type: 'radio',
-        value: 'false',
-        subscription: { value: true },
-    })
-
-    // Used for the 'clear' button, but works
-    const clearField = useFieldWithPatchedOnFocus(fieldname, {
-        type: 'radio',
-        value: '',
-        subscription: { value: true },
-    })
+    const [value, setValue] = useState(() => convertBooleanValue(initialValue))
+    const [lastSyncedValue, setLastSyncedValue] = useState(() =>
+        convertBooleanValue(initialValue)
+    )
+    const [syncTouched, setSyncTouched] = useState(false)
```

Comment:
In "genric-input" we just do `setValueSynced(value === newValue)`. What is the reason that this and other fields need the effect and `syncTouched`? Could we not calculate this? Or should the `generic-input` also use this way of doing it?

Reply:
I think it's more correct to do the side effect with useEffect. In the generic input, which is the first one I wrote, this `setValueSynced(value === newValue)` was working, but in the other components, it does not work, I suppose because the code is executing before the new value has actually been set.

Reply:
Alright, I just in general try to not use `useEffect`, and always try to look for other solutions. It adds additional re-renders and complexity. But maybe in this case it's warranted - because we're actually syncing states here. 

But I do feel like the need for "syncTouched", "lastSyncedValue", and "setValueSynced" is somewhat complex. 
Ideally this is just something we could calculate from mutation state, and potentially the query-cache. But it's hard to argue for such changes when this seems to work great.